packages/wrangler/src/versions/deploy.ts

```diff
@@ -587,9 +588,20 @@ async function maybePatchSettings(
 		},
 	});
 
+	const observability = [];
+	if (patchedSettings.observability) {
+		observability.push(`enabled: ${patchedSettings.observability.enabled}`);
+		if (patchedSettings.observability.head_sampling_rate) {
+			observability.push(
+				`head_sampling_rate: ${patchedSettings.observability.head_sampling_rate}`
+			);
+		}
+	}
 	const formattedSettings = formatLabelledValues(
 		{
 			logpush: String(patchedSettings.logpush ?? "<skipped>"),
+			observability:
+				observability.length > 0 ? observability.join("\n") : "<skipped>",
```

Comment:
if `observability` was a Record<string, string> rather than Array<string> you could nest calls to `formatLabelledValues` with the `indentationCount` options to get nice formatting

Reply:
If you have time. If not, no need to block this PR. I can come back and clean it up later.

Reply:
Done!